[#145603813] Add rc4 (de|en)cryption support

RyanMeulenkamp opened the pull request:
With this commit, the U-boot has a rc4 command that could be used to
decrypt rc4 encrypted data. To accomplish this the following was done:

* Add cmd_rc4.c, the actual functionality behind the command
* Add a reference in the makefile, to add it to the commands

jaapdejong commented:
Please add the `setexpr` command also

jaapdejong reviewed (changes requested):
Please add the setexpr command also

RyanMeulenkamp committed:
:closed_lock_with_key: [#145603813] Add rc4 (de|en)cryption support

With this commit, the U-boot has a rc4 command that could be used to
decrypt rc4 encrypted data. To accomplish this the following was done:

* Add cmd_rc4.c, the actual functionality behind the command
* Add a reference in the makefile, to add it to the commands
* The defconfig was modified to include setexpr along with some other
  changes
